webapp/src/components/Metrics/PerformanceAnalysis.test.tsx

```diff
@@ -0,0 +1,138 @@
+      availableDatasetSplits={availableDatasetSplits}
+      datasetSplitName="eval"
+      setDatasetSplitName={setDatasetSplitName}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+describe("PerformanceAnalysis", () => {
+  const handlers = [
+    getCustomMetricInfoAPIResponse,
+    getMetricsPerFilterAPIResponse,
+  ];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have two toggle buttons for the train and test", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    // verify if the toggles displayed with correct names
+    expect(
+      screen.getByRole("button", { name: "Evaluation Set", pressed: true })
+    ).toHaveValue("eval");
+    expect(
+      screen.getByRole("button", { name: "Training Set", pressed: false })
+    ).toHaveValue("train");
+  });
+
+  it("should display the expected columns", async () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    await waitFor(() => {
+      const expectedColumnHeaders = [
+        "Label",
+        "Total",
+        "Correct & Predicted",
+        "Correct & Rejected",
+        "Incorrect & Rejected",
+        "Incorrect & Predicted",
+        "Accuracy",
+        "Precision",
+        "Recall",
+        "F1",
+        "ECE",
+      ];
+      const actualColumnHeaders = screen.getAllByRole("columnheader");
+      expectedColumnHeaders.forEach((name, index) => {
+        name.includes("Correct")
+          ? expect(screen.getAllByTestId("CheckIcon")).toBeTruthy()
+          : name.includes("Incorrect")
+          ? expect(screen.getAllByTestId("XIcon")).toBeTruthy()
+          : expect(actualColumnHeaders[index].textContent).toEqual(name);
+      });
+    });
+  });
+
+  it("should display the default sort as desc for colum header 'Total number of utterances'", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    const header = screen.getByRole("columnheader", { name: /Total/ });
+    expect(header).toHaveAttribute("aria-sort", "descending");
+  });
+
+  it("should display the list of options if the user click on the column header 'filterValue'", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    fireEvent.mouseDown(screen.getByRole("button", { name: "Label" }));
+    const filterList = screen.getAllByRole("option");
+    expect(filterList).toHaveLength(10);
+
+    const expectedList = [
+      "Label",
+      "Prediction",
+      "Smart Tags",
+      "Extreme Length",
+      "Partial Syntax",
+      "Dissimilar",
+      "Almost Correct",
+      "Behavioral Testing",
+      "Pipeline Comparison",
+      "Uncertain",
+    ];
+    filterList.forEach((item, index) => {
+      expect(item.textContent).toEqual(expectedList[index]);
+    });
+
+    expect(screen.getByRole("option", { name: "Smart Tags" })).toHaveAttribute(
```

Comment:
Instead of `Smart Tags`, maybe we should check for `Prediction` since we click on it after? This was, we make sure it was not already selected.
```suggestion
    expect(screen.getByRole("option", { name: "Prediction" })).toHaveAttribute(
```

Reply:
Smart tags should always be unselected, So I added the test case to ensure it. We can add more test cases to verify the prediction aria-selected before here. Is this what you suggest here?